src/System.Memory/src/System/Span.cs

```diff
@@ -233,6 +233,210 @@ internal Span(Pinnable<T> pinnable, IntPtr byteOffset, int length)
+                    Unsafe.As<byte, IntPtr>(ref Unsafe.Add<byte>(ref b, i + 4 * ptrSize)) = default(IntPtr);
+                    Unsafe.As<byte, IntPtr>(ref Unsafe.Add<byte>(ref b, i + 5 * ptrSize)) = default(IntPtr);
+                    Unsafe.As<byte, IntPtr>(ref Unsafe.Add<byte>(ref b, i + 6 * ptrSize)) = default(IntPtr);
+                    Unsafe.As<byte, IntPtr>(ref Unsafe.Add<byte>(ref b, i + 7 * ptrSize)) = default(IntPtr);
+                    i += ptrSize * 8;
+                }
+                while (i < (byteLength - ptrSize))
+                {
+                    Unsafe.As<byte, IntPtr>(ref Unsafe.Add<byte>(ref b, i)) = default(IntPtr);
+                    i += ptrSize;
+                }
+            }
+            while (i < byteLength)
+            {
+                Unsafe.Add<byte>(ref b, i) = 0;
+                ++i;
+            }
+            #endregion
+            #region Other memset like impl
+            //size_t blockIdx;
+            //size_t blocks = count >> 3;
+            //size_t bytesLeft = count - (blocks << 3);
+            //_UINT64 cUll =
+            //    c
+            //    | (((_UINT64)c) << 8)
+            //    | (((_UINT64)c) << 16)
+            //    | (((_UINT64)c) << 24)
+            //    | (((_UINT64)c) << 32)
+            //    | (((_UINT64)c) << 40)
+            //    | (((_UINT64)c) << 48)
+            //    | (((_UINT64)c) << 56);
+
+            //_UINT64* destPtr8 = (_UINT64*)dest;
+            //for (blockIdx = 0; blockIdx < blocks; blockIdx++) destPtr8[blockIdx] = cUll;
+
+            //if (!bytesLeft) return dest;
+
+            //blocks = bytesLeft >> 2;
+            //bytesLeft = bytesLeft - (blocks << 2);
+
+            //_UINT32* destPtr4 = (_UINT32*)&destPtr8[blockIdx];
+            //for (blockIdx = 0; blockIdx < blocks; blockIdx++) destPtr4[blockIdx] = (_UINT32)cUll;
+
+            //if (!bytesLeft) return dest;
+
+            //blocks = bytesLeft >> 1;
+            //bytesLeft = bytesLeft - (blocks << 1);
+
+            //_UINT16* destPtr2 = (_UINT16*)&destPtr4[blockIdx];
+            //for (blockIdx = 0; blockIdx < blocks; blockIdx++) destPtr2[blockIdx] = (_UINT16)cUll;
+
+            //if (!bytesLeft) return dest;
+
+            //_UINT8* destPtr1 = (_UINT8*)&destPtr2[blockIdx];
+            //for (blockIdx = 0; blockIdx < bytesLeft; blockIdx++) destPtr1[blockIdx] = (_UINT8)cUll;
+
+            //return dest;
+            #endregion
+            #region custom align, stride in type size impl
+            #endregion
+        }
+
+        /// <summary>
+        /// Fills the contents of this span with the given value.
+        /// </summary>
+        public void Fill(T value)
+        {
+            if (_length == 0) { return; }
+
+            ref T r = ref DangerousGetPinnableReference();
+            int length = _length;
+            // TODO: Remove ugly regions when the right impl has been found
+            #region Naive impl
```

Comment:
@jkotas Curious, why `InitBlockUnaligned` instead of `InitBlock`? Aren't spans of 1-byte types always going to be aligned?